### PR TITLE
feat(TDP-3502): allow to specify mongodb.uri as a whole instead of specifics parts

### PR DESCRIPTION
**Pull request by hmdebenque**
https://jira.talendforge.org/browse/TDP-3502

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**(Optional) Other information**:
- add encryption capabilities for passwords in URIs
- enable password encryption in mongodb.uri
- use mongodb.uri property instead of port,user,...

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1138/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1138/)

File: dataprep-backend-service/src/main/java/org/talend/dataprep/encrypt/AESEncryption.java [hunk 1]
```diff
@@ -88,6 +92,66 @@ public static String decrypt(final String src) throws Exception {
         return new String(decValue, ENCODING);
     }
 
+    public static String decryptUriPassword(final String rawUri) {
```
Comment (fhuaulme) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/encrypt/AESEncryption.java]: Missing javadoc

File: dataprep-backend-service/src/main/java/org/talend/dataprep/encrypt/PropertiesEncryption.java [hunk 2]
```diff
@@ -119,12 +129,27 @@ private String encryptIfNot(String input) {
...
+            } catch (Exception e) {
+                // Property was already decrypted
+                LOGGER.debug("Trying to decrypt a non encrypted property.", e);
+                return input;
+            }
         }
     }
+
+    private static boolean isUri(String field) {
+        UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS + UrlValidator.ALLOW_ALL_SCHEMES + UrlValidator.ALLOW_2_SLASHES);
```
Comment (fhuaulme) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/encrypt/PropertiesEncryption.java]: Code format.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/13/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/13/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/14/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/14/)

**Pull request merged**